### PR TITLE
[rocm-comgr] fix path to linker

### PR DESCRIPTION
**Pull request by guiguid**

### CODE REVIEW

**Comment by acxz** [comment]
Thx for the PR @guiguid! However it is not completely up to spec. Can up update the `.SRCINFO` as well? See: https://wiki.archlinux.org/index.php/.SRCINFO

**Comment by guiguid** [comment]
Done. Sorry for the missing update of .SRCINFO.

**Comment by jmert** [comment]
This patch didn't work for me as is. Instead, I had to change it to:
```diff
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -21,6 +21,7 @@ build() {
         -DClang_DIR=/opt/rocm/lib/cmake/clang \
         -DLLD_INCLUDE_DIRS=/opt/rocm/include/lld \
         -DROCM_DIR=/opt/rocm/share/rocm/cmake \
+        -DCMAKE_SHARED_LINKER_FLAGS='-L/opt/rocm/lib -Wl,-rpath-link,/opt/rocm/lib' \
         "$_dirname/lib/comgr"
   make
```
before `makepkg` would complete without an error.

**Comment by guiguid** [comment]
Thanks
Now it 's completely fixed !

File: rocm-comgr/PKGBUILD [hunk 1]
```diff
@@ -21,6 +21,7 @@ build() {
         -DClang_DIR=/opt/rocm/lib/cmake/clang \
         -DLLD_INCLUDE_DIRS=/opt/rocm/include/lld \
         -DROCM_DIR=/opt/rocm/share/rocm/cmake \
+        -DCMAKE_SHARED_LINKER_FLAGS_INIT='-L/opt/rocm/lib -Wl,-rpath-link,/opt/rocm/lib' \
```
Comment (ranisalt) [inline rocm-comgr/PKGBUILD:24]: Please use `CMAKE_EXE_LINKER_FLAGS` instead, `CMAKE_SHARED_LINKER_FLAGS_INIT` is meant to be set by toolchain configuration, not our use case here.

**Comment by jmert** [comment]
I added it because the compile failed with liblldELF not being found without it.

In the mean time, I've updated to Arch's llvm/clang v10 and had a reboot cycle (for unrelated reasons). Again attempting to compile without `-L/opt/rocm/lib` the build fails, but I get a new failure — [shown in this gist](https://gist.github.com/jmert/de776eb8c8795f339bc6574f1ecb8d56) — which now looks like it's finding my system liblldELF instead of the one provided by llvm-roc.

For completeness, the full linker command I found in `CMakeFiles/amd_comgr.dir/link.txt` is:
```
/opt/rocm/bin/clang++ -fPIC -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -Wl,-rpath-link,/opt/rocm/lib -shared -Wl,-soname,libamd_comgr.so.1 -o libamd_comgr.so.1.6 CMakeFiles/amd_comgr.dir/src/comgr-compiler.cpp.o CMakeFiles/amd_comgr.dir/src/comgr-device-libs.cpp.o CMakeFiles/amd_comgr.dir/src/comgr-disassembly.cpp.o CMakeFiles/amd_comgr.dir/src/comgr-elfdump.cpp.o CMakeFiles/amd_comgr.dir/src/comgr-env.cpp.o CMakeFiles/amd_comgr.dir/src/comgr-metadata.cpp.o CMakeFiles/amd_comgr.dir/src/comgr-objdump.cpp.o CMakeFiles/amd_comgr.dir/src/comgr-signal.cpp.o CMakeFiles/amd_comgr.dir/src/comgr-symbol.cpp.o CMakeFiles/amd_comgr.dir/src/comgr.cpp.o  -pthread -Wl,--version-script=/home/justin/.cache/yay/rocm-comgr/src/ROCm-CompilerSupport-rocm-3.3.0/lib/comgr/src/exportmap -Wl,--no-undefined /opt/rocm/lib/libclangFrontendTool.a -llldELF -llldCommon /opt/rocm/lib/libclangCodeGen.a /opt/rocm/lib/libclangRewriteFrontend.a /opt/rocm/lib/libclangARCMigrate.a /opt/rocm/lib/libclangStaticAnalyzerFrontend.a /opt/rocm/lib/libclangStaticAnalyzerCheckers.a /opt/rocm/lib/libclangStaticAnalyzerCore.a /opt/rocm/lib/libclangCrossTU.a /opt/rocm/lib/libclangIndex.a /opt/rocm/lib/libclangFrontend.a /opt/rocm/lib/libclangDriver.a -lpthread /opt/rocm/lib/libclangParse.a /opt/rocm/lib/libclangSerialization.a /opt/rocm/lib/libclangSema.a /opt/rocm/lib/libclangAnalysis.a /opt/rocm/lib/libclangASTMatchers.a /opt/rocm/lib/libclangEdit.a /opt/rocm/lib/libclangFormat.a /opt/rocm/lib/libclangToolingInclusions.a /opt/rocm/lib/libclangToolingCore.a /opt/rocm/lib/libclangAST.a /opt/rocm/lib/libclangRewrite.a /opt/rocm/lib/libclangLex.a /opt/rocm/lib/libclangBasic.a /opt/rocm/lib/libLLVM-11git.so 
```

**Comment by ranisalt** [comment]
It seems to me that your environment has an issue.

I am [building in a clean Arch Linux](https://github.com/ranisalt/rocm-arch/runs/604307223) container [with `-Wl,-rpath-link,/opt/rocm/lib` only](https://github.com/ranisalt/rocm-arch/blob/actions/rocm-comgr/PKGBUILD#L18) and it works.

**Comment by jmert** [comment]
I don't find any differences in the cmake output up to the point where my compilation fails.

The one thing I could not find in your build log is any installation of the regular `clang` and associated packages which are present on my machine. My errors specifically reference missing symbols in `/usr/lib64/gcc/x86_64-pc-linux-gnu/9.3.0/../../../../lib64/liblldELF.a` == `/usr/lib/liblldELF.a`, so apparently the compiler is searching `/usr/lib` before it looks in `/opt/rocm/lib`.

After uninstalling `lld-10.0.0-2` from the standard repos, the compilation succeeded with your `PKGBUILD`, so it appears there's an error in the library search priority of the ROCm builds if you happen to have the rest of the LLVM toolchain installed locally.

**Comment by ranisalt** [comment]
I tried installing `lld`, `clang` and `llvm` from the standard repos, and now I can't compile even with this patch.

**Pull request merged**